data.sql

```diff
@@ -1,10 +1,77 @@
 /* Populate database with sample data. */
 
-INSERT INTO animals
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
 VALUES(1, 'Agumon', '3-feb-2020', 0, true, 10.23);
-INSERT INTO animals
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
 VALUES(2, 'Gabumon', '15-nov-2018', 2, true, 8);
-INSERT INTO animals
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
 VALUES(3, 'Pikachu', '7-jan-2021', 1, false, 15.04);
-INSERT INTO animals
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
 VALUES(4, 'Devimon', '12-may-2017', 5, true, 11);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES(5, 'Charmander', '8-feb-2020', 0, false, -11);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES(6, 'Plantmon', '15-nov-2021', 2, true, -5.7);  
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES(7, 'Squirtle', '2-apr-1993', 3, false, -12.13);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES(8, 'Angemon', '12-jun-2005', 1, true, -45);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES(9, 'Boarmon', '7-jun-2005', 7, true, 20.4);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES(10, 'Blossom', '13-oct-1998', 3, true, 17);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES(11, 'Ditto', '14-may-2022', 4, true, 22);
+
+/* TRANSACTIONS */
+
+/* Setting species to unspecified */
+
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK;
+
+/* Setting species properly */
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon%';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+/* DELETE ALL RECORDS */
+
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+/*  Do multiple actions in one transaction */
+
+BEGIN;
+
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01'; 
+
+SAVEPOINT DELETE_AFTER_DATE_1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO DELETE_AFTER_DATE_1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+COMMIT;
```

Comment:
- [ ] Please note, transaction queries should be in the 'queries.sql` , verify changes were made after the transaction and also verify changes persist after commit

Reply:
Understood. Thanks!